st2common/tests/unit/test_metrics.py

```diff
@@ -337,3 +339,55 @@ def _get_tested(metrics_timer=None):
             test_key,
             (end_time - start_time).total_seconds()
         )
+
+
+class TestFormatMetrics(unittest2.TestCase):
+    def test_format_metrics_liveaction_db_without_key(self):
+        pack = 'test'
+        action = 'lakface'
+
+        liveaction_db = MagicMock()
+        liveaction_db.context = {'pack': pack}
```

Comment:
Would also be good to test edge cases aka when pack is not set and we default to unknown, etc.

Besides that, LGTM

Reply:
Good call, forgot about that branch. Added.